src/riot-rs-runqueue/src/runqueue.rs

```diff
@@ -135,6 +133,12 @@ mod clist {
         next_idxs: [u8; N_THREADS],
     }
 
+    impl<const N_QUEUES: usize, const N_THREADS: usize> Default for CList<N_QUEUES, N_THREADS> {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+
     impl<const N_QUEUES: usize, const N_THREADS: usize> CList<N_QUEUES, N_THREADS> {
         pub const fn new() -> Self {
             // TODO: ensure N fits in u8
```

Comment:
Actually, I think a normal `assert!(N_QUEUES < 255)` works as well. Yes, would be good to have that both for `N_QUEUES` and `N_THREADS`.
> And if so, why are these not u8 in the first place?

It must be an `usize` because it's used as array-size in the `CList` properties. I not sure if it's possible to make it an `u8` and cast it into `usize` somehow for the array?

Reply:
keep [this](https://github.com/future-proof-iot/RIOT-rs/blob/559d92fe50008b4eaa86c4a5bc9a9114e12d8484/src/riot-rs-runqueue/src/runqueue.rs#L50) in mind, for `N_THREADS`, `0xff` (255) is a reserved value.

Reply:
So the generic nature of runqueue was at some point limited by what hacspec / hax did support. Not sure if it helps but we might want to make "usize" a generic parameter?
`sentinel()` could return `N_THREADS_TYPE.max() -1` etc.

btw, `const { assert!(N_QUEUES != Self::sentinel()); }` is incorrect, only `N_THREADS` need to be `!= sentinel()`.

Reply:
ah, and let's not forget this assumption: "runqueue numbers fit in usize bits (supporting max 32 priority levels)".

Reply:
> ah, and let's not forget this assumption: "runqueue numbers fit in usize bits (supporting max 32 priority levels)"

I guess we can add `assert!(N_QUEUES < usize::BITS)`?

Reply:
`assert!(N_QUEUES <= usize::BITS)` :)